src/main/java/core/basesyntax/model/Action.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.model;
+
+public enum Action {
+    BALANCE("b"),
+    SUPPLY("s"),
+    PURCHASE("p"),
+    RETURN("r");
+
+    private String code;
+
+    Action(String code) {
+        this.code = code;
+    }
+
+    public String getCode() {
+        return code;
+    }
+
+    public static Action fromCode(String code) {
+        for (Action action : values()) {
+            if (action.getCode().equals(code)) {
+                return action;
+            }
+        }
+        throw new IllegalArgumentException("Invalid operation code: " + code);
```

Comment:
Consider using a more specific exception than `IllegalArgumentException`, such as a custom exception that indicates the type of error more clearly. This can improve error handling and make the code more understandable.